vhdbuilder/packer/test/windows-files-check.ps1

```diff
@@ -179,4 +217,152 @@ function Test-ValidateSinglePackageSignature {
     }
 }
 
-Test-ValidateAllSignature
\ No newline at end of file
+function Test-ValidateSingleFileOnMoonCake {
+    param (
+        $dir
+    )
+
+    $excludeHashComparisionListInAzureChinaCloud = @(
+        "calico-windows",
+        "azure-vnet-cni-singletenancy-windows-amd64",
+        "azure-vnet-cni-singletenancy-swift-windows-amd64",
+        "azure-vnet-cni-singletenancy-overlay-windows-amd64",
+        # We need upstream's help to republish this package. Before that, it does not impact functionality and 1.26 is only in public preview
+        # so we can ignore the different hash values.
+        "v1.26.0-1int.zip"
+    )
+
+    foreach ($URL in $map[$dir]) {
+        $fileName = [IO.Path]::GetFileName($URL)
+        $dest = [IO.Path]::Combine($dir, $fileName)
+        if (!(Test-Path $dir)) {
+            New-Item -ItemType Directory $dir -Force | Out-Null
+        }
+
+        DownloadFileWithRetry -URL $URL -Dest $dest -redactUrl
+        $globalFileSize = (Get-Item $dest).length
+        
+        $isIgnore=$False
+        foreach($excludePackage in $excludeHashComparisionListInAzureChinaCloud) {
+            if ($URL.Contains($excludePackage)) {
+                $isIgnore=$true
+                break
+            }
+        }
+        if ($isIgnore) {
+            continue
+        }
+
+        if ($URL.StartsWith("https://acs-mirror.azureedge.net/")) {
+            $mcURL = $URL.replace("https://acs-mirror.azureedge.net/", "https://kubernetesartifacts.blob.core.chinacloudapi.cn/")
+
+            $mooncakeFileSize = (Invoke-WebRequest $mcURL -UseBasicParsing -Method Head).Headers.'Content-Length'
+
+            if ($globalFileSize -ne $mooncakeFileSize) {
+                $MisMatchFile[$URL]=$mcURL
+            }
+        }
+    }
+}
+
+function Test-ValidateFilesOnMoonCake {
```

Comment:
This validates files in both global and mooncake so the function name may not be accurate.